analyzers/tests/SonarAnalyzer.UnitTest/Helpers/NavigationHelperTest.cs

```diff
@@ -65,16 +70,39 @@ public void SwitchMethod()
     }
 }";
 
+        private const string SourceTopLevelStatement = @"
+var a = 1;
+var b = 2;
+if (a == b)
+{
+    DoSomething();
+}
+if (a == 2)
+{
+    DoSomething();
+}
+void DoSomething() { }";
+
         private MethodDeclarationSyntax ifMethod;
         private MethodDeclarationSyntax switchMethod;
 
+        private SyntaxTree syntaxTreeTopLevelStatement;
+
         [TestInitialize]
         public void TestSetup()
         {
             var syntaxTree = CSharpSyntaxTree.ParseText(Source);
+            syntaxTreeTopLevelStatement = CSharpSyntaxTree.ParseText(SourceTopLevelStatement);
```

Comment:
TestInitialize is run before each test and this field is used in only one of them. We should move this to `GetPrecedingStatementTopLevelStatements`.